### DISCUSSION TITLE
package:add("deps") 能否控制直接依赖的 include, 而不加 links

### DISCUSSION THREAD

**Discussion by fasiondog** [xmake-io/xmake#3656] [Q&A]
在使用 boost.python 时，因为会自动加入 python 的依赖，导致 target 会链接 libpython3.x.so（实际不需要，只要链接 boost_python）。

但这样，在使用 anaconda 时，相应的 python target 在 引入时会报错：Not found libpython3.x.so.

如果在 PATH 里手工添加 conda libpython 的路径，可以导入，但会导致其他的 cpp 库出现失败，比如依赖的 PyQt 无法使用。

所以，希望在使用 boost.python 的时候，on_load 里只添加 python 的 include 路径，而不链接python.so。

我尝试以下两种方法修改 on_load 但都失败了：

1. package:add("deps", "python", { configs={links=""}})   没用

2. package:add("includedirs", python_include_path, package:installdir("include"))  
   这样做，在 target 里实际添加的 includedir 前面会被加上 package 自身的安装路径，导致安装失败。（不知道算不算bug）

不知道，有什么办法可以做到只添加依赖的include，而不链接相应的库？

**Comment by xq114** [comment]
package:add("deps", "python", {kind = "binary"})

**Reply by xq114** [reply]
不过要用头文件的话这个也不对，应该修改python包加一个headeronly的config。
理论上一个py module要么是和python版本无关的，这样头文件和库都不需要；要么是和python版本相关的，那应该链接到具体版本的python库，比如在conda环境下运行xmake，xmake应该会自动使用anaconda里的python，这样也不会有运行时冲突。

**Reply by waruqi** [reply]
> 应该修改python包加一个headeronly的config。

嗯 可以试下这个

**Reply by fasiondog** [reply]
pybind是要链接到python.so, 但链接到boost_python的target不能再链接到 python.so, 反而会有问题。而且，不是很清楚 boost.python 是怎么做到的，如果使用静态链接的boost.python,  在 linux 下即使boost.python链接的是不匹配版本的 python，也不会有任何问题，交叉编译也是如此。我先试下 headeronly
